File: Resources/Projects/Archive/2023/Competition/Programming/Contest/TOI-Zero_68/A2/019/TOI-Zero_68/A3/020/020.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int h1,h2,b1,b2,x,y;
    cin >> h1 >> h2 >> b1 >> b2 >> x >> y;
    int mx=0;
    for(int i=0;i<=x;i++){
        for(int j=0;j<=x-i;j++){
            for(int a=0;a<=y;a++){
                for(int b=0;b<=y-a;b++){
                    int hh1=i+b,hh2=j+a;
                    int bb1=i+a,bb2=j+b;
                    if(hh1<=h1 && hh2<=h2 && bb1<=b1 && bb2<=b2){
                    	mx=max(mx,i+j+a+b);
					}
                }
            }
        }
    }
    cout << mx;
}

File: TOI-Zero_68/A3/020/020.py
h1, h2, b1, b2 = map(int, input().split())
x, y = map(int, input().split())

max_soldiers = 0

for i in range(x + 1):
    for j in range(x - i + 1):
        for a in range(y + 1):
            for b in range(y - a + 1):
                hh1 = i + b
                hh2 = j + a
                bb1 = i + a
                bb2 = j + b
                
                if hh1 <= h1 and hh2 <= h2 and bb1 <= b1 and bb2 <= b2:
                    max_soldiers = max(max_soldiers, i + j + a + b)

print(max_soldiers) 
